src/scaffoldmaker/meshtypes/meshtype_3d_stellate1.py

```diff
@@ -166,70 +225,81 @@ def generateBaseMesh(cls, region, options):
+            markerGroup = findOrCreateFieldGroup(fm, "marker")
+            markerName = findOrCreateFieldStoredString(fm, name="marker_name")
+            markerLocation = findOrCreateFieldStoredMeshLocation(fm, mesh, name="marker_location")
+
+            markerPoints = findOrCreateFieldNodeGroup(markerGroup, nodes).getNodesetGroup()
+            markerTemplateInternal = nodes.createNodetemplate()
+            markerTemplateInternal.defineField(markerName)
+            markerTemplateInternal.defineField(markerLocation)
 
         # Create nodes
-        numNodesPerArm = [0]
-        dipMultiplier = 1
         nodeIdentifier = 1
+        numNodesPerArm = [0]
         minArmAngle = 2 * math.pi / armCount
-        halfArmArcAngleRadians = minArmAngle/2
-        xx = []
-        xds1 = []
-        xds2 = []
-        x_in_nodes = []
-        for na in range(armCount):
-            elementsCount_i = [elementsCount1[na], elementsCount2, elementsCount3]
-            x, ds1, ds2, nWheelEdge = createArm(halfArmArcAngleRadians, elementLengths, elementLengthCentral, elementsCount_i, dipMultiplier, armCount, na)
-            for ix in range(len(x)):
-                if na == 0 or ix not in nWheelEdge:
-                    node = nodes.createNode(nodeIdentifier, nodetemplate)
-                    cache.setNode(node)
-                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_VALUE, 1, x[ix])
-                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS1, 1, ds1[ix])
-                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS2, 1, ds2[ix])
-
-                    nodeIdentifier += 1
-                    x_in_nodes.append(x[ix])
-            numNodesPerArm.append(len(x))
-            xx.append(x)
-            xds1.append(ds1)
-            xds2.append(ds2)
+        halfArmArcAngleRadians = minArmAngle / 2
+        if not isMeanMouse:
+            dipMultiplier = 1
+            for na in range(armCount):
+                elementsCount_i = [elementsCount1[na], elementsCount2, elementsCount3]
+                x, ds1, ds2, nWheelEdge = createArm(halfArmArcAngleRadians, elementLengths, elementLengthCentral, elementsCount_i, dipMultiplier, armCount, na)
+                for ix in range(len(x)):
+                    if na == 0 or ix not in nWheelEdge:
+                        node = nodes.createNode(nodeIdentifier, nodetemplate)
+                        cache.setNode(node)
+                        coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_VALUE, 1, x[ix])
+                        coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS1, 1, ds1[ix])
+                        coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS2, 1, ds2[ix])
+                        nodeIdentifier += 1
+                numNodesPerArm.append(len(x))
+        else:
+            numNodesPerArm = options['numNodesPerArm']
+            x_dx_all = options['meanMouseMeshNodes']
```

Comment:
Don't get these values from options. Can get these values directly since you know Base parameter set and have gleaned 'isMouse' and 'isMean'.
I think we can guarantee that the mean mouse has the same numbers of nodes, elements as the non-mean. I would prefer that the same code is used to calculate numNodesPerArm for all cases. In the generated nodes case you use len(x), but it should be computable from elementsCounts along arms.
Also rename to nodesCountsPerArm for consistency with Count being used by other variables.